mmv1/third_party/terraform/resources/resource_bigtable_table.go

```diff
@@ -105,42 +106,39 @@ func resourceBigtableTableCreate(d *schema.ResourceData, meta interface{}) error
 
 	defer c.Close()
 
-	name := d.Get("name").(string)
+	tableId := d.Get("name").(string)
+	tblConf := bigtable.TableConf{TableID: tableId}
+
+	// Set the split keys if given.
 	if v, ok := d.GetOk("split_keys"); ok {
-		splitKeys := convertStringArr(v.([]interface{}))
-		// This method may return before the table's creation is complete - we may need to wait until
-		// it exists in the future.
-		err = c.CreatePresplitTable(ctx, name, splitKeys)
-		if err != nil {
-			return fmt.Errorf("Error creating presplit table. %s", err)
-		}
-	} else {
-		// This method may return before the table's creation is complete - we may need to wait until
-		// it exists in the future.
-		err = c.CreateTable(ctx, name)
-		if err != nil {
-			return fmt.Errorf("Error creating table. %s", err)
-		}
+		tblConf.SplitKeys = convertStringArr(v.([]interface{}))
 	}
 
+	// Set the column families if given.
+	columnFamilies := make(map[string]bigtable.GCPolicy)
 	if d.Get("column_family.#").(int) > 0 {
 		columns := d.Get("column_family").(*schema.Set).List()
 
 		for _, co := range columns {
 			column := co.(map[string]interface{})
 
 			if v, ok := column["family"]; ok {
-				// Set a longer timeout as adding column family can be slow. The current timeout
-				// in the Go client is less than one minute. This timeout should not be longer
-				// than the overall timeout.
-				ctxWithTimeout, cancel := context.WithTimeout(ctx, 5*time.Minute)
-				defer cancel() // Always call cancel.
-				if err := c.CreateColumnFamily(ctxWithTimeout, name, v.(string)); err != nil {
-					return fmt.Errorf("Error creating column family %s. %s", v, err)
-				}
+				// By default, there is no GC rules.
+				columnFamilies[v.(string)] = bigtable.NoGcPolicy()
 			}
 		}
 	}
+	tblConf.Families = columnFamilies
+
+	// This method may return before the table's creation is complete - we may need to wait until
+	// it exists in the future.
+	// Set a longer timeout as creating table and adding column families can be pretty slow.
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, 20*time.Minute)
```

Comment:
NIT: You can keep this as just `ctx` I believe.

Reply:
ack